src/pytorch_lightning/callbacks/model_summary.py

```diff
@@ -62,6 +65,13 @@ def on_fit_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -
         if trainer.is_global_zero:
             self.summarize(summary_data, total_parameters, trainable_parameters, model_size)
 
+    def _summary(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> Union[DeepSpeedSummary, Summary]:
+        from pytorch_lightning.strategies.deepspeed import DeepSpeedStrategy
+
+        if isinstance(trainer.strategy, DeepSpeedStrategy) and trainer.strategy.zero_stage_3:
+            return DeepSpeedSummary(pl_module, max_depth=self._max_depth)
```

Comment:
maybe I am missing something. Does it return an instance instead of the summary?

Reply:
that seems to be correct. the `summarize` function below also returns an instance of the summary